proof_key.go

```diff
@@ -95,12 +100,32 @@ func (proof *KeyAbsentProof) Verify(key, value []byte, root []byte) error {
 
 // Bytes returns a go-wire binary serialization
 func (proof *KeyAbsentProof) Bytes() []byte {
-	return wire.BinaryBytes(proof)
+	return append([]byte{keyAbsentMagicNumber}, wire.BinaryBytes(proof)...)
 }
 
-// ReadKeyAbsentProof will deserialize a KeyAbsentProof from bytes.
-func ReadKeyAbsentProof(data []byte) (*KeyAbsentProof, error) {
+// readKeyAbsentProof will deserialize a KeyAbsentProof from bytes.
+func readKeyAbsentProof(data []byte) (*KeyAbsentProof, error) {
 	proof := new(KeyAbsentProof)
 	err := wire.ReadBinaryBytes(data, &proof)
 	return proof, err
 }
+
+func ReadKeyProof(data []byte) (KeyProof, error) {
```

Comment:
In the `Bytes()` method, you serialize a byte slice with go wire, then prepend a type.

Why not just do the same here?

```
b, val := data[0], data[1:]
switch b {
case keyExistsMagicNumber:
 return readKeyExistsProof(val)
...
```

Reply:
Heh yeah, that's much simpler.